src/screens/ChangeForgotPasswordScreen.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from 'react';
+import { Keyboard, Text, TouchableOpacity, TouchableWithoutFeedback, View } from 'react-native';
+import TextInputCustom from '../components/common/TextInput';
+import { useNavigation } from '@react-navigation/native';
+import { faEye, faEyeSlash } from '../constants/icons/FontAwesome';
+import { BackgroundPaper } from '../components/BackgroundPaper';
+import { styles } from '../themes/appTheme';
+import { PasswordValidationMsg } from '../components/PasswordValidationMsg';
+import { useForm } from '../hooks/useForm';
+import { useAppDispatch } from '../store/hooks';
+import { addToast } from '../store/feature/toast/toastSlice';
+import { StatusType } from '../types/common';
+import { ArrowBack } from '../components/ArrowBack';
+import { BigTitle } from '../components/BigTitle';
+import useSpikyService from '../hooks/useSpikyService';
+
+const initialState = {
+    newPassword: '',
+    confirmPassword: '',
+};
+
+export const ChangeForgotPasswordScreen = ({ route }: { route: any }) => {
+    // deep link stuff
+    const params = route.params || {};
+    const { token, correoValid } = params;
+    // end deep link stuff
+    const dispatch = useAppDispatch();
+    const { updatePasswordUri } = useSpikyService();
+    const [buttonState, setButtonState] = useState(false);
+    const [passVisible1, setPassVisible1] = useState(true);
+    const [passVisible2, setPassVisible2] = useState(true);
+    const [msgPassword, setMsgPassword] = useState(false);
+    const [passwordValid, setPasswordValid] = useState(false);
+    const navigation = useNavigation<any>();
+    const { form, onChange } = useForm(initialState);
+
+    const { newPassword, confirmPassword } = form;
+
+    const changePassword = async () => {
+        if (passwordValid && newPassword === confirmPassword) {
+            try {
+                await updatePasswordUri(token, correoValid, newPassword);
+                onChange(initialState);
+                navigation.navigate('LoginScreen');
+            } catch (error) {
+                console.log(error);
+                dispatch(addToast({ message: 'Cambio no completado', type: StatusType.WARNING }));
+            }
+        } else if (!passwordValid) {
+            dispatch(
+                addToast({
+                    message: 'La contraseña no cumple los criterios',
+                    type: StatusType.WARNING,
+                })
+            );
+        } else if (newPassword !== confirmPassword) {
+            dispatch(
+                addToast({
+                    message: 'Las contraseñas no coinciden',
+                    type: StatusType.WARNING,
+                })
+            );
+        }
+    };
```

Comment:
Podria ser mejor asi?

```TS    
const validatePassword = (): Toast | undefined => {
        if (!passwordValid) {
            return {
                message: 'La contraseña no cumple los criterios',
                type: StatusType.WARNING,
            };
        }
        if (newPassword !== confirmPassword) {
            return {
                message: 'Las contraseñas no coinciden',
                type: StatusType.WARNING,
            };
        }
        return undefined;
    };

    const changePassword = async () => {
        const passwordValidated = validatePassword();
        if (passwordValidated === undefined) {
            try {
                await updatePassword(uid, currentPassword, newPassword);
                onChange(initialState);
                navigation.navigate('ConfigurationScreen');
            } catch (error) {
                console.log(error);
                dispatch(addToast({ message: 'Contraseña incorrecta', type: StatusType.WARNING }));
            }
        } else {
            dispatch(addToast(passwordValidated));
        }
    };
```

Reply:
Love it! Donde ponen uds este tipo de funciones? En `utils/` o en `helpers/` o prefieres en otro?

Reply:
No estoy seguro, igual si no hay carpeta especifica estaria bueno tenerla.